sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetQuerySuite.scala

```diff
@@ -229,4 +229,81 @@ class ParquetQuerySuite extends QueryTest with ParquetTest with SharedSQLContext
       }
     }
   }
+
+  test("SPARK-10301 Clipping nested structs in requested schema") {
```

Comment:
Can we list all cases that are tested at here? Cases that should be here will be
* two struct types have the same fields
* two struct types have totally two different set of fields
* one struct type is a super set of another one
* there are some common fields. But, there are also fields that only exist in one file. I believe that the ordering of fields is also matter at here. For example, for a struct in the global schema, if its fields are `a, b, c, d`, `a, d` in local struct and `a, b` in struct filed are two different cases.